### PR TITLE
CUMULUS-2898 - patch1

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2898](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2898)

This PR pulls in #3556 

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:50]: We're removing the `improveStackTrace` call here. What's the reasoning? Is it no longer necessary with the v3 functions?

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:103]: Same question re: `improveStackTrace`. I think all of the formatting changes here are just updating the indenting because we removed this call.

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:139]: Do we have some auto-formatting happening here? Might be worth a quick convo.

File: packages/aws-client/tests/test-StepFunctions.js
Comment (npauzenga) [inline packages/aws-client/tests/test-StepFunctions.js:205]: Are these new tests?

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:21]: I removed this export, looks like it may not be necessary.

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:45]: thanks, using strongly typed objects is good

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:50]: Per Charles and Jenny, v3 has better stack track built in, so they have been removing "improveStackTrace" in their v3 updates.

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:103]: Per Charles and Jenny, v3 has better stack track built in, so they have been removing "improveStackTrace" in their v3 updates.

File: packages/aws-client/tests/test-StepFunctions.js
Comment (tclark-innovim) [reply packages/aws-client/tests/test-StepFunctions.js:205]: Yes, I added these unit tests to provide 100% coverage for StepFunctions.ts. Is that ok?

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:139]: i'll connect with you on slack

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [reply packages/aws-client/src/StepFunctions.ts:50]: Neat! We can probably remove that method once everything is switched over 🤔

File: packages/aws-client/tests/test-StepFunctions.js
Comment (npauzenga) [reply packages/aws-client/tests/test-StepFunctions.js:205]: That's great! Just went over them. Thanks for doing that!

**Review by npauzenga — APPROVED** [review]
Looks awesome @tclark-innovim. Thanks for picking this up and adding extra coverage 💪

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:21]: Looks like a duplicate import

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:46]: nit: as long as you're being helpful in describing the actual return type, a description would be excellent.

e.g.
`@returns {Promise<DescribeExecutionOutput>} An object representing a step function name and id` (or whatever it actually is)

We don't own `DescribeExecutionOutput` so explaining it is arguably not necessary and you could just do 

`@returns {Promise<DescribeExecutionOutput>}`

Obviously super minor and in no way a blocking comment 😄

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:65]: Same as above re: docstrings. Take it or leave it.

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [inline packages/aws-client/src/StepFunctions.ts:113]: Presumably `events` can be null or undefined? If so 👍

File: packages/aws-client/tests/test-StepFunctions.js
Comment (npauzenga) [inline packages/aws-client/tests/test-StepFunctions.js:21]: Don't need this? What replaces it? 

(Likely about to answer my own question reading on)

Comment (npauzenga) [reply packages/aws-client/tests/test-StepFunctions.js:21]: Is it just that we're more easily able to sub AWS functions? Don't need the throttling here?

File: packages/message/src/StepFunctions.ts
Comment (npauzenga) [inline packages/message/src/StepFunctions.ts:53]: Can we be more specific? Not your change so won't hold up the PR.

File: packages/message/src/StepFunctions.ts
Comment (npauzenga) [inline packages/message/src/StepFunctions.ts:95]: Can we be more specific? Not your change so won't hold up the PR.

File: packages/message/src/StepFunctions.ts
Comment (npauzenga) [inline packages/message/src/StepFunctions.ts:161]: Why the cast?

**Review by npauzenga — CHANGES_REQUESTED** [review]
Looks great @tclark-innovim! Just a couple, mostly clarifying comments.

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:21]: Lines 7 and 12 are imports, but Line 21 is an export. I believe Charles originally added this line, so that they type can be used by another package (orca-recovery-adapter) that depends on aws-client.

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:46]: noted, I removed the type name from the return description

File: packages/aws-client/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/aws-client/src/StepFunctions.ts:65]: noted, I removed the type name from the return description

File: packages/message/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/message/src/StepFunctions.ts:95]: thanks

File: packages/message/src/StepFunctions.ts
Comment (tclark-innovim) [reply packages/message/src/StepFunctions.ts:161]: event.type is a HistoryEvent. In order to compare it to a list of strings, the cast is necessary. I think this was also from Charles, but that is my understanding of why.

File: packages/aws-client/tests/test-StepFunctions.js
Comment (tclark-innovim) [reply packages/aws-client/tests/test-StepFunctions.js:21]: Well, the tests that use this runWithStubbedAndThrottledSfnOperation function were updated to throttle inside each test, so this function became unused, and was removed.

File: packages/aws-client/src/StepFunctions.ts
Comment (npauzenga) [reply packages/aws-client/src/StepFunctions.ts:21]: Oh geez. Sorry, can't read.

**Review by npauzenga — APPROVED** [review]
Looks great @tclark-innovim! Thanks again for navigating this. Approved pending CI.

**Pull request merged**
